pypesto/collections/collections.py

```diff
@@ -0,0 +1,484 @@
+                 x_vectors: np.ndarray,
+                 x_names: Sequence[str] = None,
+                 vector_tags: Tuple[int, int] = None,
+                 coll_type: CollectionType = None,
+                 predictions: Sequence[CollectionPrediction] = None,
+                 lower_bound: np.ndarray = None,
+                 upper_bound: np.ndarray = None):
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        x_vectors:
+            parameter vectors of the collection, in the format
+            n_parameter x n_vectors
+        x_names:
+            Names or identifiers of the parameters
+        vector_tags:
+            Additional tag, which adds information about the the parameter
+            vectors of the form (optimization_run, optimization_step) if the
+            collection is created from an optimization result or
+            (sampling_chain, sampling_step) if the collection is created from a
+            sampling result.
+        coll_type:
+            Type of collection: Ensemble (default) or sample
+            Samples are meant to be representative, ensembles can be any
+            collection of parameters
+        predictions:
+            List of CollectionPrediction objects
+        lower_bound:
+            array of potential lower bounds for the parameters
+        upper_bound:
+            array of potential upper bounds for the parameters
+        """
+
+        # handle parameter vectors and sizes
+        self.x_vectors = x_vectors
+        self.n_x = x_vectors.shape[0]
+        self.n_vectors = x_vectors.shape[1]
+        self.vector_tags = vector_tags
+        self.summary = None
+
+        # store bounds
+        self.lower_bound = np.full((self.n_x,), np.nan)
+        if lower_bound is not None:
+            self.lower_bound = lower_bound
+        self.upper_bound = np.full((self.n_x,), np.nan)
+        if upper_bound is not None:
+            self.upper_bound = upper_bound
+
+        # handle parameter names
+        if x_names is not None:
+            self.x_names = x_names
+        else:
+            self.x_names = [f'x_{ix}' for ix in range(self.n_x)]
+
+        # Do we have a representative sample or just random ensemble?
+        self.coll_type = CollectionType.ensemble
+        if coll_type is not None:
+            self.coll_type = coll_type
+
+        # Do we have predictions for this ensemble?
+        self.predictions = []
+        if predictions is not None:
+            self.predictions = predictions
+
+    def __iter__(self):
+        yield X_VECTOR, self.x_vectors
+        yield NX, self.n_x
+        yield X_NAMES, self.x_names
+        yield NVECTORS, self.n_vectors
+        yield VECTOR_TAGS, self.vector_tags
+        yield COLLECTION_TYPE, self.coll_type
+        yield PREDICTIONS, self.predictions
+        yield SUMMARY, self.summary
+        yield LOWER_BOUND, self.lower_bound
+        yield UPPER_BOUND, self.upper_bound
+
+    def predict(self,
```

Comment:
Could move prediction bounds to this method, and remove from `CollectionPrediction`